source/main.cc

```diff
@@ -641,65 +652,21 @@ int main (int argc, char *argv[])
       // the parameter file
       possibly_load_shared_libs (input_as_string);
 
-      // now switch between the templates that code for 2d or 3d. it
-      // would be nicer if we didn't have to duplicate code, but the
-      // following needs to be known at compile time whereas the dimensionality
-      // is only read at run-time
-      ParameterHandler prm;
-
+      // Now switch between the templates that start the model for 2d or 3d.
       switch (dim)
         {
           case 2:
           {
-            aspect::Simulator<2>::declare_parameters(prm);
-            parse_parameters (input_as_string, prm);
-
-            if (output_xml)
-              {
-                if (i_am_proc_0)
-                  prm.print_parameters(std::cout, ParameterHandler::XML);
-              }
-            else if (output_plugin_graph)
-              {
-                aspect::Simulator<2> flow_problem(MPI_COMM_WORLD, prm);
-                if (i_am_proc_0)
-                  flow_problem.write_plugin_graph (std::cout);
-              }
-            else
-              {
-                aspect::Simulator<2> flow_problem(MPI_COMM_WORLD, prm);
-                flow_problem.run();
-              }
+            run_simulator<2>(input_as_string,output_xml,output_plugin_graph);
             break;
           }
-
           case 3:
           {
-            aspect::Simulator<3>::declare_parameters(prm);
-            parse_parameters (input_as_string, prm);
-
-            if (output_xml)
-              {
-                if (i_am_proc_0)
-                  prm.print_parameters(std::cout, ParameterHandler::XML);
-              }
-            else if (output_plugin_graph)
-              {
-                aspect::Simulator<3> flow_problem(MPI_COMM_WORLD, prm);
-                if (i_am_proc_0)
-                  flow_problem.write_plugin_graph (std::cout);
-              }
-            else
-              {
-                aspect::Simulator<3> flow_problem(MPI_COMM_WORLD, prm);
-                flow_problem.run();
-              }
-
+            run_simulator<3>(input_as_string,output_xml,output_plugin_graph);
             break;
           }
-
           default:
-            AssertThrow((dim >= 2) && (dim <= 3),
+            AssertThrow(false,
                         ExcMessage ("ASPECT can only be run in 2d and 3d but a "
                                     "different space dimension is given in the parameter file."));
```

Comment:
This is where the difference in output comes from. I'd say to just revert to the previous condition -- I know it's duplicative, but it's also instructive.